core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/PostingsHighlighter.java

```diff
@@ -44,13 +46,15 @@
 import java.util.Locale;
 import java.util.Map;
 
+@Deprecated
 public class PostingsHighlighter implements Highlighter {
+    private static DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(PostingsHighlighter.class));
 
     private static final String CACHE_KEY = "highlight-postings";
 
     @Override
     public HighlightField highlight(HighlighterContext highlighterContext) {
-
+        deprecationLogger.deprecated("[postings] highlighter is deprecated, please use [unified] instead");
```

Comment:
this way we will issue a deprecation warning for each hit, each field. Should we try to do it for each document at least? or maybe we can do it for each request, that would be even better.